platforms/nuttx/src/px4/stm/stm32f4/px4io_serial/px4io_serial.cpp

```diff
@@ -432,10 +434,10 @@ ArchPX4IOSerial::_do_interrupt()
 			if ((length < 1) || (length < PKT_SIZE(*_current_packet))) {
 				perf_count(_pc_badidle);
 
-				/* stop the receive DMA */
-				stm32_dmastop(_rx_dma);
+				/* abort dma on error */
+				_abort_dma();
 
-				/* complete the short reception */
+				/* error out */
```

Comment:
```suggestion
				/* complete the short reception indicated the error */
```

Reply:
This is an error case, no? We get into that state when the interrupt is called, but there are still bytes missing, so it's more an error than completion.

Reply:
It should say something like /* This is a malformed packet, complete the reception indicating the error */

Reply:
Done